### PR TITLE
Add Helix to Editor Support

### PR DESCRIPTION
**Pull request by AlexanderBrevig**
Helix has built in support of elvish, and it uses `elvish -lsp` as its LSP provider.
https://github.com/helix-editor/helix/blob/master/languages.toml#L1522

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +40,7 @@ A curated list of awesome Elvish packages, modules, and tools that support Elvis
 | [Sublime Text 3](https://www.sublimetext.com) | [elvish_syntax_for_sublime](https://github.com/href/elvish_syntax_for_sublime) |
 | [PyCharm](https://www.jetbrains.com/pycharm) | [elvish-lang-plugin](https://github.com/sblundy/elvish-lang-plugin) |
 | [Micro](https://github.com/zyedidia/micro) | [elvish-micro-syntax](https://github.com/kolbycrouch/elvish-micro-syntax) |
+| [Helix](https://github.com/helix-editor/helix) | Builtin support; [elvish](https://github.com/elves/elvish) as `elvish -lsp` |
```
Comment (xiaq) [inline README.md:43]: I can't quite parse "elvish as `elvish -lsp`"; can you elaborate what that means?

Comment (AlexanderBrevig) [reply README.md:43]: I mean literally, just running `elvish -lsp`

From help:
```
elvish --help | rg "lsp|language"
  -lsp
    	Run the builtin language server
```

😃

Comment (AlexanderBrevig) [reply README.md:43]: Oh sorry, you mean the grammar is bad. 
Maybe just 
> Builtin support with `elvish -lsp` 

?

Comment (xiaq) [reply README.md:43]: I see, that's clearer. I'll change to that wording.

**Pull request merged**